rust/cubestore/cubestore/src/queryplanner/projection_above_limit.rs

```diff
@@ -107,15 +107,38 @@ fn projection_above_limit(plan: &LogicalPlan) -> Result<LogicalPlan> {
     }
 }
 
-struct ColumnRecorder {
-    columns: HashSet<Column>,
+/// A `Vec<Column>` -- or, when we don't need that, a `()`.
+trait ColumnCollector {
+    fn push(&mut self, column: &Column);
 }
 
-impl ExpressionVisitor for ColumnRecorder {
+impl ColumnCollector for () {
+    fn push(&mut self, _column: &Column) {}
+}
+
+impl ColumnCollector for Vec<Column> {
+    fn push(&mut self, column: &Column) {
+        self.push(column.clone());
+    }
+}
+
+#[derive(Default)]
+struct ColumnRecorder<T: ColumnCollector> {
+    column_hash: HashSet<Column>,
```

Comment:
@srh, what do you think about using [`indexmap`](https://github.com/indexmap-rs/indexmap)?

Reply:
Sure.